lib/compute/jenkins-main-node.ts

```diff
@@ -361,6 +360,22 @@ export class JenkinsMainNode {
     ];
   }
 
+  /** Creates the commands to install plugins, typically done in blocks */
+  public static createPluginInstallCommands(pluginList: string[]): InitCommand[] {
+    const pluginInstallBlockSize = 10;
+    const jenkinsCliCommand = `java -jar /jenkins-cli.jar -s http://localhost:8080 -auth @${JenkinsMainNode.JENKINS_DEFAULT_ID_PASS_PATH}`;
+    const pluginListCopy = Object.assign([], pluginList);
+    const pluginListSlices: String[] = [];
+    do {
+      pluginListSlices.push(pluginListCopy.splice(0, pluginInstallBlockSize).join(' '));
+    } while (pluginListCopy.length !== 0);
+
+    return pluginListSlices.map((slice, index) => {
+      const extraCommand = (index === pluginListSlices.length - 1) ? `${jenkinsCliCommand} restart` : '';
```

Comment:
Can you explain why a restart is needed?

Reply:
After all the plugins are installed jenkins is restarted, this was done before on line 334, which was removed

Reply:
Got it.